wa/framework/workload.py

```diff
@@ -218,7 +229,25 @@ def teardown(self, context):
 
     @once_per_instance
     def finalize(self, context):
-        pass
+        self.remove_assets(context)
+
+    def deploy_assets(self, context, target_directory=None):
+        if not self.deployable_assets:
+            return
+
+        if not target_directory:
+            target_directory = self.target.working_directory
+        self.target.execute('mkdir -p {}'.format(target_directory))
+
+        for i, asset in enumerate(self.deployable_assets):
+            self.target.push(self.asset_files[i], target_directory)
+            self.deployed_assets.append(self.target.path.join(target_directory,
```

Comment:
The join could be replaced with `self.target.get_workpath(asset)`

Reply:
This was left as join to allow for custom directories.